crates/solidity/testing/snapshots/bindings_output/yul/slot_suffix/generated/0.4.11-success.txt

```diff
@@ -0,0 +1,20 @@
+# This file is generated automatically by infrastructure scripts. Please don't edit by hand.
+
+References and definitions: 
+   ╭─[input.sol:1:1]
+   │
+ 1 │ contract Test {
+   │          ──┬─  
+   │            ╰─── def: 1
+ 2 │     function test(bytes storage data) public {
+   │              ──┬─               ──┬─  
+   │                ╰────────────────────── def: 2
+   │                                   │   
+   │                                   ╰─── def: 3
+   │ 
+ 4 │             let s := sload(data_slot)
```

Comment:
shouldn't this be resolved to a `slot` property in the built-ins? rather than the `data` parameter?

Reply:
We don't have a built-ins struct to represent the `.offset` and `.slot` for Yul. I'll add it. But in this case I'm not sure, because `data_slot` is equivalent to `data.slot` so there are two bindings there. I think it may be confusing to bind this to the built-in and not `data`.

Reply:
I added a built-in struct to define the possible members of Yul external values. Binding to it is hard-coded in the rules, whenever we encounter a `YulPath` with a member access.

One I noticed is that the definition of `YulPath` does not have version restrictions, but it's only available since 0.7.0 (from what I can test with Remix). Is this worth addressing in the grammar?

Reply:
> because `data_slot` is equivalent to `data.slot` so there are two bindings there.

Yes, and IIUC, both should be bound to the `slot` built-in property (of "int" type), right? not `data` (of type `bytes`).
WDYT?

> the definition of `YulPath` does not have version restrictions, but it's only available since `0.7.0` 

`let x.y.z := 0` is legal in `0.6.12`. Am I missing something?